jlm/llvm/opt/alias-analyses/ModRefSummary.hpp

```diff
@@ -27,74 +26,59 @@ class ModRefSummary
   [[nodiscard]] virtual const PointsToGraph &
   GetPointsToGraph() const noexcept = 0;
 
-  [[nodiscard]] virtual const jlm::util::HashSet<const PointsToGraph::MemoryNode *> &
-  GetRegionEntryNodes(const rvsdg::Region & region) const = 0;
-
-  [[nodiscard]] virtual const jlm::util::HashSet<const PointsToGraph::MemoryNode *> &
-  GetRegionExitNodes(const rvsdg::Region & region) const = 0;
-
-  [[nodiscard]] virtual const jlm::util::HashSet<const PointsToGraph::MemoryNode *> &
-  GetCallEntryNodes(const rvsdg::SimpleNode & callNode) const = 0;
-
-  [[nodiscard]] virtual const jlm::util::HashSet<const PointsToGraph::MemoryNode *> &
-  GetCallExitNodes(const rvsdg::SimpleNode & callNode) const = 0;
-
   /**
-   * Retrieves the set of memory locations that may be targeted by the given pointer typed value
-   * @param output the output producing the pointer value
-   * @return a conservative set of memory locations the pointer may target
+   * Provides the set of MemoryNodes that represent memory locations that may be
+   * modified or referenced by the given simple node.
+   *
+   * The simple node can be any operation that affects memory:
+   *  - Load, Store, Memcpy, Free, Call, which have memory states routed through them
+   *  - Alloca or Malloc, which produce memory states
```

Comment:
This list is non-exhaustive. So I would rather give examples, and also reference the proper operations with \ref